Use dagger 2.0

ligi opened the pull request:
And move Constants to central place + cleanup

cbalster commented:
Not tested yet, but looks good so far. One thing though:
I really, really dislike dumping all constants - without any logical connection - into a single class. There is absolutely no reason to do so as they all have a perfectly fine place to live in that corresponds to their use. For example the http timeout should just be moved to the component imho.

ligi commented:
Removed the constants in central position - I like this though - can give you a good entry-point to new developers and forces you to expose magical constants - the logical connection is done by the source of them - but nothing I have a strong oppinion though - and nothing that should block this PR so I removed it

cbalster commented:
:+1: 
I think we should refactor everything possible to use constructor injection to explicitly state the dependencies. Can be done in a follow-up PR though.
